src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -149,21 +139,34 @@
           error-in-words?          (or (not last-partial-word-valid?)
                                        (not butlast-words-valid?))
           upper-case?              (boolean (re-find #"[A-Z]" @seed-phrase))
+          error                    (rf/sub [:enter-seed-phrase/error])
```

Comment:
What is the rationale to prefer the state in the app-db instead of a local state? The local state wouldn't need to cater for unmounting situations. If the error is mostly limited to this component, and also prop drilling wouldn't be an issue, a local state can be better.

Reply:
Hi @ilmotta,

Once the seed phrase is validated, we dispatch the on-success function. This function further validates based on the flow. For example:

- In the account creation flow, the key-uid of the entered seed phrase should not be present in the profiles.
- In the backup keycard flow, the key-uid of the entered seed phrase should match the key-uid of the current profile.

Since the validations are happening outside of this screen, it seems simpler to store the error state in the app-db. I could alternatively pass the set-error function, but we’re not supposed to call functions from events. Therefore, I need to create an effect and pass this function to that event.

Reply:
> Since the validations are happening outside of this screen,

By the way, I also tried moving the flow-based validation into the screen itself, but it seemed more cluttered to me.

https://github.com/status-im/status-mobile/pull/22252#discussion_r1986971590

Reply:
UPD: Renamed error to global-error